### PR TITLE
[compute] Proximity placement group support

### PR DESCRIPTION
**Pull request by adewaleo**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

File: src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_validators.py [hunk 1]
```diff
@@ -68,6 +68,26 @@ def validate_keyvault(cmd, namespace):
...
 
+def validate_proximity_placement_group(cmd, namespace):
+    from msrestazure.tools import parse_resource_id, is_valid_resource_id
+
+    namespace.proximity_placement_group = _get_resource_id(cmd.cli_ctx, namespace.proximity_placement_group,
+                                                           namespace.resource_group_name,
+                                                           'proximityPlacementGroups', 'Microsoft.Compute')
+
+    if not namespace.proximity_placement_group or not is_valid_resource_id(namespace.proximity_placement_group):
+        logger.debug('Proximity placement group is %s', namespace.proximity_placement_group)
```
Comment (yugangw-msft) [inline src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_validators.py]: Is this a correct trace message when the PPD is not available?

File: src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_validators.py [hunk 2]
```diff
@@ -68,6 +68,26 @@ def validate_keyvault(cmd, namespace):
...
+        return
+
+    parsed = parse_resource_id(namespace.proximity_placement_group)
+
+    rg = parsed['resource_group']
+    name = parsed['name']
+
+    if not check_existence(cmd.cli_ctx, name, rg, 'Microsoft.Compute', 'proximityPlacementGroups'):
```
Comment (yugangw-msft) [inline src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_validators.py]: what will happen at service end, if we don't do the check at client end? Do we get a good error?

File: src/command_modules/azure-cli-vm/setup.py [hunk 3]
```diff
@@ -34,7 +34,7 @@
 DEPENDENCIES = [
     'azure-mgmt-msi==0.2.0',
     'azure-mgmt-authorization==0.50.0',
-    'azure-mgmt-compute==4.6.1',
+    'azure-mgmt-compute==4.6.2',
```
Comment (yugangw-msft) [inline src/command_modules/azure-cli-vm/setup.py]: you might need to update  other command  modules such as acs

File: src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_validators.py (hunk 1 above)
Comment (adewaleo) [reply src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_validators.py]: This is just for debugging purpose to show what was processed for the parameter. I can reword the logging message.

File: src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_validators.py (hunk 2 above)
Comment (adewaleo) [reply src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_validators.py]: I believe the error should be something like the following, if the ppg is in a different region or does not exist:

```
Deployment failed. Correlation ID: 260fc599-b6f4-418f-9c2e-fc800602d78f. {
  "error": {
    "code": "NotFound",
    "message": "Proximity Placement Group '/subscriptions/00977cdb-163f-435f-9c32-39ec8ae61f4d/resourceGroups/ova-test/providers/Microsoft.Compute/proximityPlacementGroups/tosinppg' cannot be found."
  }
}
```

I can remove this existence check, however the CLI does perform this check for availability set...

File: src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_validators.py [hunk 4]
```diff
@@ -68,6 +68,26 @@ def validate_keyvault(cmd, namespace):
...
+        raise CLIError("Availability set '{}' does not exist.".format(name))
```
Comment (adewaleo) [inline src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_validators.py]: This should be proximity placement group not availability set

File: src/command_modules/azure-cli-vm/setup.py (hunk 3 above)
Comment (adewaleo) [reply src/command_modules/azure-cli-vm/setup.py]: Yes @yugangw-msft acs and servicefabric have been updated.

File: src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_validators.py [hunk 5]
```diff
@@ -76,7 +76,7 @@ def validate_proximity_placement_group(cmd, namespace):
                                                            'proximityPlacementGroups', 'Microsoft.Compute')
 
     if not namespace.proximity_placement_group or not is_valid_resource_id(namespace.proximity_placement_group):
-        logger.debug('Proximity placement group is %s', namespace.proximity_placement_group)
+        logger.debug('namespace.proximity_placement_group is %s', namespace.proximity_placement_group)
```
Comment (tjprescott) [inline src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_validators.py]: Why is this here?

Comment (adewaleo) [reply src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_validators.py]: debug statement just in case the user enters an invalid proximity placement group. The previous lines try to parse the resource if and return None. Maybe I should actually raise a CLI error if this fails to parse...

What do you think?

File: src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_params.py [hunk 6]
```diff
@@ -50,13 +51,24 @@ def load_arguments(self, _):
 
     extension_instance_name_type = CLIArgumentType(help="Name of the vm's instance of the extension. Default: name of the extension.")
 
-    if StorageAccountTypes:
-        disk_sku = CLIArgumentType(arg_type=get_enum_type(StorageAccountTypes))
+    # StorageAccountTypes renamed to DiskStorageAccountTypes in 2018_06_01 of azure-mgmt-compute
+    if StorageAccountTypes and not DiskStorageAccountTypes:
+        DiskStorageAccountTypes = StorageAccountTypes
```
Comment (yugangw-msft) [inline src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_params.py]: maybe `DiskStorageAccountTypes  = DiskStorageAccountTypes or StorageAccountTypes`
